jaeger/spans.go

```diff
@@ -0,0 +1,121 @@
+package jaeger
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+	"sort"
+	"strconv"
+	"time"
+
+	jaegerModels "github.com/jaegertracing/jaeger/model/json"
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/log"
+)
+
+// TODO / Question: make limit configurable? Selected from UI?
+const tracesLimit = 100
+
+type Span struct {
+	jaegerModels.Span
+	TraceSize int `json:"traceSize"`
+}
+
+func getSpans(client http.Client, endpoint *url.URL, namespace, service, startMicros, endMicros string) ([]Span, error) {
+	u := endpoint
+	queryService := service
+	if config.Get().ExternalServices.Tracing.NamespaceSelector {
+		queryService = service + "." + namespace
+	}
+	u.Path = path.Join(u.Path, "/api/traces")
+	q := u.Query()
+	q.Set("service", queryService)
+	q.Set("start", startMicros)
+	q.Set("end", endMicros)
+	q.Set("limit", strconv.Itoa(tracesLimit))
+
+	u.RawQuery = q.Encode()
+	traces, _, err := queryTraces(client, u)
+	if err != nil {
+		return []Span{}, err
+	}
+
+	spans := tracesToSpans(traces, service, namespace)
+	if len(traces) == tracesLimit {
+		// Reached the limit, trying to be smart enough to show more and get the most relevant ones
+		log.Info("Limit of traces was reached, trying to find more relevant spans...")
+		return findRelevantSpans(client, spans, u, q, service, namespace)
+	}
+
+	return spans, nil
+}
+
+func tracesToSpans(traces []jaegerModels.Trace, service, namespace string) []Span {
+	spans := []Span{}
+	for _, trace := range traces {
+		// First, get the desired processes for our service
+		processes := make(map[jaegerModels.ProcessID]bool)
+		for pId, process := range trace.Processes {
+			if process.ServiceName == service || process.ServiceName == service+"."+namespace {
```

Comment:
minor, I'd add a local variable like in https://github.com/kiali/kiali/pull/1994/files#diff-f5d5eff94f17a5f6374cb3b80d4608e2R30 as it's same concept here.